ui/components/src/LineChart/LineChart.tsx

```diff
@@ -100,7 +101,18 @@ export function LineChart({ height, data, unit, grid, legend, visualMap, onDataZ
     enableDataZoom(chartRef.current);
   }
 
+  const handleOnClick = () => {
+    if (pinTooltip === false) {
+      setPinTooltip(true);
+    } else {
+      setPinTooltip(false);
+    }
```

Comment:
Nit: I think you want to use the callback form of `setState` here just to be 100% sure you're always seeing the latest state, e.g.:

```
setPinTooltip(current => !current);
```